include/deal.II/matrix_free/tools.h

```diff
@@ -119,8 +146,381 @@ namespace MatrixFreeTools
       additional_data.mapping_update_flags_inner_faces |
       additional_data.mapping_update_flags_boundary_faces;
   }
-} // namespace MatrixFreeTools
 
+  namespace internal
+  {
+    template <typename Number>
+    struct LocalCSR
+    {
+      std::vector<unsigned int> row_lid_to_gid;
+      std::vector<unsigned int> row{0};
```

Comment:
Can you remove the initialization from the struct declaration and initialize at first use instead? Alternatively use a constructor.